src/body/tee_body.rs

```diff
@@ -0,0 +1,116 @@
+use bytes::Bytes;
+use http_body::{Body, Frame, SizeHint};
+use std::fmt::{self, Debug, Formatter};
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::AsyncWrite;
+
+use super::BoxBody;
+use crate::error::DynError;
+
+// This struct needs refactored to contain a state enum to ensure that
+// we're able to (flush if necessary) and shutdown the sink when body
+// stops producing frames...
+pub struct TeeBody {
+    body: BoxBody,
+    sink: Pin<Box<dyn AsyncWrite + Send + Sync>>,
+    next: Option<Bytes>,
+}
+
+impl TeeBody {
+    pub fn new<T, U>(body: T, sink: U) -> Self
+    where
+        T: Body<Data = Bytes, Error = DynError> + Send + Sync + 'static,
+        U: AsyncWrite + Send + Sync + 'static,
+    {
+        Self {
+            body: BoxBody::new(body),
+            sink: Box::pin(sink),
+            next: None,
+        }
+    }
+
+    #[inline]
+    pub(crate) fn cap(self) -> BoxBody {
+        // I'm considering bringing back a custom impl of `BoxBody` in the
+        // event that the struct wrapper breaks composition. It's possible
+        // that we'll want to project
+        todo!()
+    }
+}
+
+impl Body for TeeBody {
+    type Data = Bytes;
+    type Error = DynError;
+
+    fn poll_frame(
```

Comment:
This is still a rough draft. I'm going to think about this much more. No rush to get this feature out.

Reply:
Lot's more work to do here. I'm enjoying taking my time to think about the API that makes the most sense.